### DISCUSSION TITLE
Microsoft released BitNet inference code bitnet.cpp, shows huge improvement over llama.cpp (100B at 7 token/s on a single M2 Ultra!)

### DISCUSSION THREAD

**Discussion by Dampfinchen** [ggml-org/llama.cpp#9945] [Ideas]
https://github.com/microsoft/BitNet

This is just insanity. 
![m2_performance](https://github.com/user-attachments/assets/a58454bd-cdfb-4e78-b572-f0304831ff83)

![intel_performance](https://github.com/user-attachments/assets/b059980b-4061-4081-8781-25f3caca6cdb)

**Comment by myan-o** [comment]
Will it be integrated?

**Comment by iplayfast** [comment]
https://www.youtube.com/watch?v=Ilna0soNn9I
